### ISSUE TITLE
Cannot define function in mu-template file

### ISSUE THREAD

**Issue by GoogleCodeExporter** [byzhang/lh-vim#29]
```
What steps will reproduce the problem?
1. write the following template file
VimL: function! s:echo(str)
VimL:   echo a:str
VimL: endfunction
2. save it in after/template/echo.template
3. execute :MuTemplate echo

What is the expected output?
No error

What do you see instead?
Error detected while processing function 
<SNR>49_TemplateAndJump..<SNR>49_Template..<SNR>49_InterpretLines..<SNR>49_Inter
pretCommand:
line    6:
E126: Missing :endfunction


What version of the product are you using? On what operating system?
Linux Ubuntu 8.04
VIM - Vi IMproved 7.0 (2006 May 7, compiled Jun 12 2009 07:07:08)
Included patches: 1, 3-4, 7-9, 11, 13-17, 19-26, 29-31, 34-44, 47, 50-56, 
58-64, 66-73, 75, 77-92, 94-107, 109, 202, 234-237
Mu-Template 2.0.3

Please provide any additional information below.


```

Original issue reported on code.google.com by `corrado9...@gmail.com` on 27 Aug 2010 at 11:33

**Comment by GoogleCodeExporter** [comment]
```
Fixing this issue will require to recognize and extract function definitions. 

Investigations result: The definition cannot be stored into a temporary file 
which will be sourced eventually, as the resulting function, if made 
script-local, would never to visible to the current template-file.
However, if a multi-lines string is built as long as endf\%[unction] is not 
seen, the string should be usable with :exe.



By the mean time, I only have one workaround that consists in defining new 
functions in (autoload) plugins. I haven't found how to support functions 
defined on one unique line.

I also see a way to emulate functions through template-file inclusions, but 
it's kind of clumsy.
```

Original comment by `luc.herm...@gmail.com` on 27 Aug 2010 at 1:13
- Changed state: **Accepted**
- Added labels: **muTemplate**

**Comment by GoogleCodeExporter** [comment]
```
Fixed in version 2.0.4
```

Original comment by `luc.herm...@gmail.com` on 19 Sep 2010 at 10:58
- Changed state: **Fixed**
